server.py

```diff
@@ -1,806 +1 @@
-#!/usr/local/bin/python3
```

Comment:
Should probably leave the shebang in for compatibility. Other than that, this is a sold PR that addresses many issues.

Reply:
Great point, thanks. I am new to Python development, so still getting the hang of all this. 😄